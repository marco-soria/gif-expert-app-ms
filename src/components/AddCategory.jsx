import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');
    /* const handleInputChange = (e) => {
        setInputValue(e.target.value);
      } */
    const handleInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        
        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

  return (
    <form action="" onSubmit={handleSubmit} aria-label="form">
      <input type="text" 
              placeholder='Search...'
              value={inputValue}
              // onChange={ (e) => setInputValue(e.target.value) }
              onChange={ handleInputChange }
      />
      {/* <button type="submit">Add a series</button> */}
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}



