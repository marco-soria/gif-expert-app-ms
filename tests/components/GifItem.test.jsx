import { render, screen } from "@testing-library/react";
import {GifItem} from "../../src/components/GifItem";

describe('GifItem', () => {

    const title='Satoru';
    const url= 'https://www.google.com/satoru.jpg';
    test('should match the snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the url and alt indicated', () => {
        render(<GifItem title={title} url={url}/>);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        //expect(src.replace(/\/$/, '')).toBe(url);
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show the title in the component', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });


});