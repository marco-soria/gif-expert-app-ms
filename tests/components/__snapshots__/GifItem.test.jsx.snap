// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Satoru"
      src="https://www.google.com/satoru.jpg"
    />
    <p>
      Satoru
    </p>
  </div>
</div>
`;
